src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -160,10 +160,7 @@ public synchronized void invalidate(NSpec spec) {
     ) {
       if ( ns_ instanceof NanoService ) {
         spec_ = spec;
-        Object ns = buildService(x_);
-        if ( ns instanceof FObject ) {
-          ((FObject) ns_).copyFrom((FObject) ns);
-        }
+        buildService(x_);
```

Comment:
The copyFrom was so that we could copy arguments to the running service if they changed. How is this accomplished now?

Reply:
Instead of setting the return value from buildService to the local variable ns and copyFrom the result to ns_, we directly update ns_ in buildService which gets ride of the need to do copyFrom. Am i missing something here?

Reply:
Many services may have cached the old service in instance variables, so we can't actually switch to a new object in the context because all of the cached versions will still be out there un-updated. All we can do is really update the parameters of the existing object, which is the same one that everyone already has. So, we can't really create a new one, just update the setters of the current one.

Reply:
addressed in https://github.com/kgrgreer/foam3/pull/1902/commits/fd2e96243f7019df930f2a05555044cab6befcbc